data/tools/wesnoth/wmltools3.py

```diff
@@ -1128,7 +1128,7 @@ def __init__(self, textdomain, isocode, topdir=""):
             expect = False
             fuzzy = "#, fuzzy\n"
             gettext = f.read().decode("utf8")
-            matches = re.compile("""(msgid|msgstr)((\s*".*?")+)""").finditer(gettext)
+            matches = re.compile(r"""(msgid|msgstr)((\s*".*?")+)""").finditer(gettext)
```

Comment:
What's confusing, exactly? The only quotes that are part of the string are the ones in `(\s*".*?")`. I don't see what's confusing about it.

Reply:
confusion reducing is one of use cases why both ' and " work after all

Reply:
The triple outer quotes. Does the search pattern actually has two consecutive double quotes? Is it searching for `""(msgid|msgstr)((\s*".*?")+)""`?

Reply:
No. Triple quotes in Python are a stronger form of quoting that can span multiple lines.

Reply:
If it is just searching for `(msgid|msgstr)((\s*".*?")+)`, it could be written `r"(msgid|msgstr)((\s*".*?")+)"`, but it isn't. I'm simply trying to understand why.

Reply:
I'll change it to single quotes.

Reply:
I think this could be just `r'''(msgid|msgstr)((\s*".*?")+)'''`, this allows the search string to be multiline while remove any confusion.

Reply:
If it's single quotes, do they even need to be tripled?

Reply:
A quick test shows both `r'(msgid|msgstr)((\s*".*?")+)'` and `r'''(msgid|msgstr)((\s*".*?")+)'''` work, but `r"(msgid|msgstr)((\s*".*?")+)" doesn't (probably because the search pattern has ".

Reply:
Yes, triple-quoting also means you don't need to escape quotes of the same type.

Reply:
I'm trying to run this code to verify that nothing has broken, and I'm puzzled. Where should the `file` function that's called on line 1121 be defined?

Reply:
I can't see it anywhere either, maybe its supposed to be [open](https://docs.python.org/3/library/functions.html#open) and is a typo? I guess that because it takes in a string path and returns a file object. You could try triggering that particular block to see if it throws any errors.

Reply:
I think it should be File

Reply:
> The way it handles utf-8 makes me think this is Python2 code.
`file()` is Python2, the Python3 equivalent is `open()`

Reply:
I think we mark this as resolved, and then I open a new PR saying it's dead code. Replaced by `import gettext`.